databases/mongoose.md

```diff
@@ -0,0 +1,275 @@
+# Mongoose as an ODM/ORM
+
+### Projected Time
+
+About 60 - 90 minutes
+- 15-20 minutes for Lesson
+- 15 minutes for Guided Practice
+- 30-40 minutes for Independent Practice
+- 10 minutes for Challenge
+- 15 minutes for Check for Understanding
+
+### Prerequisites
+
+- [JSON](https://github.com/Techtonica/curriculum/blob/master/api/apis-and-json.md)
+- [Node.js: Server](../node-js/node-js.md)
+- [Express.js: Framework](../express-js/express.md)
+- [MongoDB](../databases/mongo-db.md)
+
+### Motivation
+
+Not all data can be or should be stored in a relational way. In those cases, it is more efficient to use a NoSQL database. MongoDB is one of the most popular NoSQL databases. MongoDB stores data in JSON-like "documents," which is familiar to folks who use JavaScript.   
+Mongoose is the most popular ORM's available out there that help us interact with MongoDB databases using JavaScript directly. Mongoose basically helps simplify performing operations on the MongoDB data with it's builtin functions and methods and is super user-friendly in nature and very easy to use.  
+ Mongoose provides a straight-forward, schema-based solution to modeling your application data and includes built-in type casting, validation, query building, business logic hooks and more.
+
+**Which companies use Mongoose?**  
+As per the resource available at [Tech Share](https://stackshare.io/) over 132 companies use mongoose in their tech share to building real time applications.  
+Some examples includes:  
+
+- [WebbyLab](https://webbylab.com/)
+- [triGo GmbH.](https://www.trigo.at/)  
+
+Check out the full details [here](https://stackshare.io/mongoose).
+
+### Objectives
+
+**Participants will be able to:**
+- Understand what an ORM/ODM is and why it is used.
+- Understand how ORM/ODM connects the project to a Mongo DB.
+- Use mongoose as an ORM/ODM to perform basic CRUD on database.
+
+
+### Materials
+
+Some great reading resources to get you started: 
+
+- [Mongoose docs on MDN](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/mongoose)
+- [Mongoose Docs](https://mongoosejs.com/)
+
+
+### Lessons
+
+- [Understanding Mongoose as an ODM](https://code.tutsplus.com/articles/an-introduction-to-mongoose-for-mongodb-and-nodejs--cms-29527)
```

Comment:
Could you tell apprentices how long you expect each of these to take?  Do you think they should watch this entire video, or just part of it?